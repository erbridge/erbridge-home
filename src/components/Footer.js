import React from 'react';

import ExternalLink from './ExternalLink';
import EmailIcon from './EmailIcon';
import GitHubIcon from './GitHubIcon';
import LinkedInIcon from './LinkedInIcon';
import TwitterIcon from './TwitterIcon';

import './Footer.css';

const Footer = () => (
  <div className="Footer">
    <div className="Footer__content">
      <div className="Footer__contact-links">
        <ExternalLink
          className="Footer__contact-link"
          to="https://twitter.com/erbridge"
        >
          <TwitterIcon />
        </ExternalLink>
        <ExternalLink
          className="Footer__contact-link"
          to="https://github.com/erbridge"
        >
          <GitHubIcon />
        </ExternalLink>
        <ExternalLink
          className="Footer__contact-link"
          to="https://www.linkedin.com/in/erbridge"
        >
          <LinkedInIcon />
        </ExternalLink>
        <ExternalLink
          className="Footer__contact-link"
          to="mailto:hello@erbridge.co.uk"
        >
          <EmailIcon />
        </ExternalLink>
      </div>
      <div className="Footer__copyright">
        <div className="Footer__copyright-line">Â© 2015 - 2018 erbridge</div>
        <div className="Footer__copyright-line">
          <ExternalLink to="https://raw.githubusercontent.com/erbridge/erbridge-home/master/LICENSE">
            MIT
          </ExternalLink>{' '}
          (<ExternalLink to="https://github.com/erbridge/erbridge-home">
            code
          </ExternalLink>) |{' '}
          <ExternalLink to="https://creativecommons.org/licenses/by/3.0/">
            CC-BY
          </ExternalLink>{' '}
          (content)
        </div>
      </div>
    </div>
  </div>
);

export default Footer;
