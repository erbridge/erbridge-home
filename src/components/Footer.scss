@import 'styles/colours';
@import 'styles/layout';

@include wrapper-with-content(Footer);

.Footer {
  background-color: $darkest-grey;
  padding: 5px;
}

.Footer__content {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.Footer__contact-links {
  display: flex;
  margin-top: 10px;
}

.Footer__contact-link {
  width: 48px;
  margin: 2px;
}

.Footer__copyright {
  margin: 10px 0;
}

.Footer__copyright-line {
  margin: 5px 0;
  text-align: center;
}
