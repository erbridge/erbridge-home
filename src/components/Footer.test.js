import { shallow } from 'enzyme';
import React from 'react';

import Footer from './Footer';

it('renders correctly', () => {
  expect(shallow(<Footer />)).toMatchSnapshot();
});
