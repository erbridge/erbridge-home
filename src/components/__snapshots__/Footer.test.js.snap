// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Footer"
>
  <div
    className="Footer__content"
  >
    <div
      className="Footer__copyright"
    >
      <div
        className="Footer__copyright-line"
      >
        Â© 2015 - 2017 erbridge
      </div>
      <div
        className="Footer__copyright-line"
      >
        <ExternalLink
          to="https://raw.githubusercontent.com/erbridge/erbridge-home/master/LICENSE"
        >
          MIT
        </ExternalLink>
         
        (
        <ExternalLink
          to="https://github.com/erbridge/erbridge-home"
        >
          code
        </ExternalLink>
        ) |
         
        <ExternalLink
          to="http://creativecommons.org/licenses/by/3.0/"
        >
          CC-BY
        </ExternalLink>
         
        (content)
      </div>
    </div>
  </div>
</div>
`;
